Microsoft.TeamServices.Samples.Client/Release/ReleasesSample.cs

```diff
@@ -0,0 +1,512 @@
+
+        public ReleaseDefinition GetAReleaseDefinition()
+        {
+            string projectName = ClientSampleHelpers.FindAnyProject(this.Context).Name;
+
+            // Get a release client instance
+            VssConnection connection = Context.Connection;
+            ReleaseHttpClient releaseClient = connection.GetClient<ReleaseHttpClient>();
+
+            List<ReleaseDefinition> releaseDefinitions = releaseClient.GetReleaseDefinitionsAsync(project: projectName).Result;
+
+            int releaseDefinitionId = releaseDefinitions.FirstOrDefault().Id;
+
+            // Show a releaes definitions
+            ReleaseDefinition releaseDefinition = releaseClient.GetReleaseDefinitionAsync(project: projectName, definitionId: releaseDefinitionId).Result;
+
+            Console.WriteLine("{0} {1}", releaseDefinition.Id.ToString().PadLeft(6), releaseDefinition.Name);
+
+            return releaseDefinition;
+        }
+
+        [ClientSampleMethod]
+
+        public ReleaseDefinition CreateAReleaseDefinition()
+        {
+            string projectName = ClientSampleHelpers.FindAnyProject(this.Context).Name;
+
+            // Get a release client instance
+            VssConnection connection = Context.Connection;
+            ReleaseHttpClient releaseClient = connection.GetClient<ReleaseHttpClient>();
+
+            // JSON to create release definition
+            string releaseDefinitionJson = "{\"name\":\"Fabrikam-web\",\"revision\" : 1,\"environments\":[{\"name\":\"PROD\",\"deployPhases\":[{\"name\":\"Run on agent\",\"phaseType\":1,\"rank\":1, \"deploymentInput\":{\"queueId\":2}}],\"preDeployApprovals\":{\"approvals\":[{\"rank\":1,\"isAutomated\":true}]},\"postDeployApprovals\":{\"approvals\":[{\"rank\":1,\"isAutomated\":true}]},\"retentionPolicy\":{\"daysToKeep\":30,\"releasesToKeep\":3,\"retainBuild\":true}}]}";
+
+            ReleaseDefinition definition = JsonConvert.DeserializeObject<ReleaseDefinition>(releaseDefinitionJson);
+
+            // create a release definition
+            ReleaseDefinition releaseDefinition = releaseClient.CreateReleaseDefinitionAsync(project: projectName, releaseDefinition: definition).Result;
+
+            Console.WriteLine("{0} {1}", releaseDefinition.Id.ToString().PadLeft(6), releaseDefinition.Name);
+
+            return releaseDefinition;
+        }
+
+        [ClientSampleMethod]
+
+        public ReleaseDefinition UpdateAReleaseDefinition()
+        {
+            string projectName = ClientSampleHelpers.FindAnyProject(this.Context).Name;
+
+            // Get a release client instance
+            VssConnection connection = Context.Connection;
+            ReleaseHttpClient releaseClient = connection.GetClient<ReleaseHttpClient>();
+
+            List<ReleaseDefinition> releaseDefinitions = releaseClient.GetReleaseDefinitionsAsync(project: projectName, searchText: releaseDefinitionName).Result;
+            int releaseDefinitionId = releaseDefinitions.FirstOrDefault().Id;
```

Comment:
Same comment as above --- do the "update" on the newly created release definition so you avoid getting the full list of release definitions just to grab the first one. This pattern also avoids impacting a legitimate release definition if a user happens to run these sample against their real account.